src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -38,11 +38,44 @@ class DebugLogicalTreeNode(
   dmOuter: () => TLDebugModuleOuterAsync,
   dmInner: () => TLDebugModuleInnerAsync
 )(implicit val p: Parameters) extends LogicalTreeNode(() => Some(device)) {
+  /**
+   * Translate register field offsets to account for the fact that the dmOuter
+   * registers are offset from the base DebugModule address.
+   *
+   * Although the rocket-chip regmap helper takes offsets that are relative to
+   * the base of the register address block, for the Object Model we have
+   * decided to collapse the two separate DebugModule register address blocks
+   * into a single one, so we need to account for the dmOuter offset.
+   */
+  private def translateDMOuterRegisterOffsets(regMap: OMRegisterMap): OMRegisterMap = {
+    val addressSets = dmOuter().dmOuter.dmiNode.address
+    addressSets.foreach { addressSet =>
+      require(
+        addressSet.contiguous,
+        s"DebugLogicalTree address logic currently assumes contiguous AddressSets; ${addressSet} is not contiguous"
+      )
+    }
+
+    val baseAddressBytes: BigInt = addressSets.map(_.base).reduceLeft(_ min _)
+
+    println(s"baseAddressBytes: ${baseAddressBytes}")
+
+    regMap.copy(
+      registerFields = regMap.registerFields.map(
+        field => field.copy(
+          bitRange = field.bitRange.copy(
+            base = field.bitRange.base + baseAddressBytes * 8
+          )
+        )
+      )
+    )
+  }
```

Comment:
@jackkoenig, @azidar, are we using any lens library in rocket-chip? It'd be pretty convenient for expressing code like the above.